main.md

```diff
@@ -828,7 +830,131 @@ receipt of a `WWW-Authenticate: Bearer` challenge from a protected resource
 resource). The effect of this likely simplifies the logistics of phased 
 upgrades to protected resources in their support DPoP or even 
 prolonged deployments of protected resources with mixed token type support. 
-  
+
+# Authorization Server-Provided Nonce {#ASNonce}
+
+Including a nonce value contributed by the authorization server in the DPoP proof
+MAY be used by authorization servers to limit the lifetime of DPoP proofs.
+The server is in control of when to require the use of a new nonce value
+in subsequent DPoP proofs.
+Without employing such a mechanism, a malicious party controlling the client
+(including potentially the end user)
+can create DPoP proofs for use arbitrarily far in the future.
+This section specifies how server-provided nonces are used with DPoP.
+
+An authorization server MAY supply a nonce value to be included by the client
+in DPoP proofs sent to it by responding to requests not including a nonce
+with a `DPoP-Nonce` HTTP header in the response supplying a nonce value to be used
+when sending the subsequent request.
+
+For example, in response to a token request without a nonce when the authorization server requires one,
+the authorization server can respond with a `DPoP-Nonce` value such as the following to provide
+a nonce value to include in the DPoP proof:
+!---
+```
+ HTTP/1.1 400 Bad Request
+ DPoP-Nonce: eyJ7S_zG.eyJH0-Z.HX4w-7v
+
+ {
+  "error": "use_dpop_nonce"
```

Comment:
We need to define this error in the OAuth Extensions Error Registry for both `Token endpoint response` as well as for `Resource access error response`.

Reply:
true ^

Reply:
I'll address Filip's comments (thanks Filip!) then I think we should merge this and publish.

Reply:
there are a few other changes from list discussions that should get in prior to publishing